cla_public/apps/checker/cait_intervention.py

```diff
@@ -102,22 +86,54 @@ def get_cait_params(category_name, organisations, choices=[], truncate=5):  # no
                 if cycle_count < intervention_quota:
                     cycle_count = 0
                 variant = "default" if cycle_count else "variant-plain"
-                params["cait_variant"] = variant
+                self.params["cait_variant"] = variant
                 if variant != "default":
                     organisations.insert(0, links_config["cait"])
                     for org in organisations:
                         org_class = org["service_name"].replace(" ", "-").lower()
                         org.update({"classname": org_class})
-                    params["truncate"] = truncate + 1
+                    self.params["truncate"] = truncate + 1
                     journey = {"uuid": get_uuid()}
                     if len(choices) > 0:
                         journey.update({"nodes": "/".join(choices), "last_node": choices[-1]})
-                    params["cait_journey"] = journey
+                    self.params["cait_journey"] = journey
+
+    def create_css_and_js_params(self, css_config, js_config):
+        if self.params.get("info_tools"):
+            self.params["cait_css"] = css_config
+            self.params["cait_js"] = js_config
+
+
+def get_cait_params(category_name, organisations, choices=[], truncate=5):
+    params_class = CreateCaitParams()
+    params = params_class.params
+    if category_name != "Family" or request.path != "/scope/refer/family":
+        return params
+
+    try:
+        cait_intervention_config = get_config()
+
+        # Make sure any errors with the json/config do not effect the site
+
+        survey_config = cait_intervention_config.get("survey", {})
+        intervention_config = cait_intervention_config.get("intervention", {})
+        nodes_config = cait_intervention_config.get("nodes", {})
+        links_config = cait_intervention_config.get("links", {})
+        css_config = cait_intervention_config.get("css", "")
+        js_config = cait_intervention_config.get("js", "")
+
+        # Survey
+        params_class.create_cait_survey_params(survey_config, choices, nodes_config)
+
+        # CAIT link
+        params_class.create_cait_link_params(intervention_config, links_config, organisations, truncate, choices)
 
         # Additional CSS injection
-        if params.get("info_tools"):
-            params["cait_css"] = css_config
-            params["cait_js"] = js_config
+        params_class.create_css_and_js_params(css_config, js_config)
```

Comment:
Is there any need to do these separately? Could we combine them into one call? Could we do them on `init`, even?